mlrun/api/db/sqldb/db.py

```diff
@@ -818,25 +824,29 @@ def list_features(
         features_results = []
         for row in query:
             feature_record = schemas.FeatureRecord.from_orm(row.Feature)
+            feature_name = feature_record.name
+
             feature_sets = self._generate_feature_set_results_with_tags_assigned(
                 row.FeatureSet, feature_set_id_tags, tag
             )
 
-            feature = schemas.Feature(
-                name=feature_record.name,
-                value_type=feature_record.value_type,
-                labels=transform_label_list_to_dict(feature_record.labels),
-            )
+            for feature_set in feature_sets:
+                # Get the feature from the feature-set full structure, as it may contain extra fields (which are not
+                # in the DB)
```

Comment:
This is part of the main change in this PR - the FeatureSet object will contain the full JSON representation of the feature-set provided. This may include fields that are not contained in the schema. The code now uses the full JSON representation to generate the returned values, and hence can contain additional fields that were provided and returned as-is.

Reply:
Talked offline, we can "workaround" this if we would add a `_full_object` field to the features table as well
since we know feel that feature is still a sub-object of the feature-set we're ok with this depdendency, but it might be changed in the future if feature will get more functionality and become an object of its own